### PR TITLE
vertical navigation structure and styles

### PR DESCRIPTION
**Pull request by thomasmaas**
**What this PR does / why we need it**:

This PR implements a service driven navigation structure with a vertical navigation that has a PatternFly 3 look & feel.

Issues to be  tackled. Please talk with Thomas when you pick one.

Done:
- [x] fix js error in many tests caused by api switcher (@josemigallas )
- [x] transform api switcher into context switcher with global navigation feedback. (@josemigallas)
- [x] Add 3scale logomark to logo  (@thomasmaas)
- [x] Render active doc in service context when attached to service, just like applications
- [x] Fix vertical nav for safari (doesn’t fold out on hover)
- [x] Create service overview page as landing page for service with re-used content from services overview and edit functionality (@thomasmaas)
- [x] Pick icons for all main items for all vertical navs  (@thomasmaas)
- [x] Move email top menu to submenu  (@thomasmaas)
- [x] Move forum top menu to submenu (@thomasmaas)
- [x] Remove all links to admin_services_path (@Martouta)
- [x] Define order service menu (@thomasmaas)
- [x] Remove unused routes: services, analytics
- [x] Don’t redirect to services after service creation, redirect to service/dashboard instead depending 
if service has been created or will be created (@Martouta).
- [x] display service subscriptions for 1 service within service section (@Martouta)
- [x] menu toggle shouldn't get active class (it's not like a popup trigger) @thomasmaas 
- [x] Dry up nav partials @thomasmaas
- [x] Improve js of vertical nav (add some delay and fix 'flash of uncollapsed when collapsed' @josemigallas 
- [x] fix 231 failing tests (most are navigation related) @Martouta, @thomasmaas and more should help soon!
- [x] conditional links in context selector https://gist.github.com/thomasmaas/2bc34eeeb323dc880ca58574763c86a6 @josemigallas 
- [x] hover bug in nav: when hovering over multiple primary menu's and then back to current primary menu secondary menu sometimes doesn't update @josemigallas 
- [x] go over all feedback and implement what makes sense
- [x] icons shouldn't move when nav collapses (@thomasmaas )
- [x] Improve api show content @thomasmaas 
- [x] check finance in menu (switch: finance) @thomasmaas 
- [x] check bubbles
- [x] fix remaining hover issues  @josemigallas 
- [x] fix navigation feedback bug for audience > accounts > subscriptions (@Martouta )
- [x] implement breadcrumbs in as it would help navigation from leaves to trunks a lot (@hery , @macejmic , @thomasmaas , @Martouta  )
- [x] pf3 styled nav with pf4 behaviour (@thomasmaas & @josemigallas)
- [x] fix some failing tests related to active docs (tests didn't; show due to GitHub problems @Martouta)
- [x] fix some pages where no active menu is set @thomasmaas & @Martouta 
- [x] fix placement of methods bubble (nothing to fix)
- [x] check master and fix some visual errors, also probably hide dev portal settings @thomasmaas 
- [x] check last time in ie11 @thomasmaas 
- [x] write doc that explains all the changes for existing customers (@thomasmaas)

Doing:

- [ ] write empathic intercom message (@thomasmaas & Andreu)
- [ ] try moving all apps to audience space and make all apis links in dashboard stand out more.
- [ ] publish on 3scale.net for saas release
- [ ] rewrite for on prem release

To do before saas deploy:
- [ ] when forum is disabled from forum settings there is no way to re-enable it. 

To do before ER1:
- [ ] fix api widget master view on prem 

Nice to have:
- [ ] remove unneeded sublayouts
- [ ] replace h2 header for h1 in provider account pages
- [ ] clean up unused css and possibly partials
- [ ] clean up unused js (hamburger)
- [ ] Only show subscriptions in audience section when they need approval (for now they are not shown at all in audience but are shown for each service)

Next iteration:
- [ ] Audience Overview Page

### CODE REVIEW

File: app/views/shared/provider/_header.html.slim [hunk 1]
```diff
@@ -56,15 +66,24 @@ div id="user_widget"
...
+        window.toggleNavigation(e.currentTarget)
+        e.preventDefault()
+      })
+      .on('click', 'body', function (e) {
+        if (e.target.type !== 'search') {
+          window.hideAllToggleable()
+        }
+      })
+      .on('click', '.vert-nav-toggle', function (e) {
+        var shouldVertNavCollapse = !JSON.parse(store.isVerticalNavCollapsed || false)
```
Comment (thomasmaas) [inline app/views/shared/provider/_header.html.slim]: I'd say the issue comes from here @didierofrivia

**Comment by Martouta** [comment]
- _Don’t redirect to services after service creation, redirect to service/dashboard instead depending
if service has been created or will be created_ <-- Done (and merged) in https://github.com/3scale/porta/pull/143 and https://github.com/3scale/porta/pull/142
- _display service subscriptions for 1 service within service section_ <-- Done (and merged) in https://github.com/3scale/porta/pull/147
- _Remove all links to admin_services_path_ <-- Done (and merged) in https://github.com/3scale/porta/pull/144
- _Remove unused routes: services, analytics_ <-- As far as I know (correct me if I am wrong), the only route left is `/admin/stats`. I am on it 😉

**Comment by didierofrivia** [comment]
I think you guys are not running `rake webpack:production` and commiting the dev ones. since these ones are indeed minified: https://github.com/3scale/porta/blob/60ac01d0ab9d2a6b49eb2d49de9c1140e10ba455/public/packs/dashboard-dea2c6f3d821230b4acc.js

**Comment by Martouta** [comment]
@didierofrivia Check https://github.com/3scale/porta/pull/155 😉

File: app/views/provider/admin/onboarding/wizard/info/outro.html.slim [hunk 2]
```diff
@@ -22,4 +22,4 @@ ul.explain.fa-ul
       markdown:
         Start sending traffic and check out the **Analytics Section**
 
-= link_to 'Got it! Take me to my API on 3scale', admin_service_path(site_account.default_service), class: 'button button--next'
```
Comment (thomasmaas) [inline app/views/provider/admin/onboarding/wizard/info/outro.html.slim]: not sure why it was `site_account` @Martouta?

**Comment by hery** [comment]
I think you mentioned the wrong Hery here, that would be @hallelujah and not @hery.
It's fun to see there's another Hery out here though! 👋

**Comment by hallelujah** [comment]
I am confused I did not see the mention though nice to make your acquitance @hery

**Comment by Martouta** [comment]
@hallelujah 👀 😝
image

**Pull request merged**